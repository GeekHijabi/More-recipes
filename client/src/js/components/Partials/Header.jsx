import React from 'react';
import { Link } from 'react-router-dom';

export const Header = () => (
  <div>
    <nav className="navbar navbar-expand-lg navbar-light">
      <Link to="/" className="navbar-brand" href="/">
        <i className="fa fa-eercast" />
                        moreRecipe
      </Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        {localStorage.getItem('token') ? '' :
        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <Link to="/signin" className="nav-link btn" href="/signin">
            Sign in
            </Link>
          </li>
          <li className="nav-item">
            <Link to="/signup" className="nav-link btn" href="/signup">
            Sign up
            </Link>
          </li>
        </ul>
        }
      </div>
    </nav>
  </div>
);


export default Header;

