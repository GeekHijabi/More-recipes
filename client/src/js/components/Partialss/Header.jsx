import React from 'react';
import ReactDOM from 'react-dom';
import 'bootstrap/dist/css/bootstrap.css';
// import { Button } from 'reactstrap';
import '../../../styles/index.scss';
/**
 * 
 * 
 * @class Header
 * @extends {React.Component}
 */
class Header extends React.Component {
/**
* @return {dom} DomElement
*/
  render() {
    return (
      <div>
        <nav className="navbar navbar-expand-lg navbar-light">
          <a className="navbar-brand" href="#">
                    <i className="fa fa-eercast" />
                        moreRecipe
          </a>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <a className="nav-link" href="./signin.html">Sign in</a>
              </li>
              <li className="nav-item">
                <a className="nav-link btn cornflowerblue" href="./signup.html">Sign up</a>
              </li>
            </ul>
            <form className="form-inline">
              <input
                            className="form-control mr-sm-2"
                            type="text"
                            placeholder="Search"
                            aria-label="Search"
              />
            </form>
          </div>
        </nav>
      </div>
    );
  }
}

export default Header;

