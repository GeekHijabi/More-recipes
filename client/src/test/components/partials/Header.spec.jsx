import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Header from '../../../js/components/Partials/Header';

configure({ adapter: new Adapter() });

describe('RecipeHeader', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
