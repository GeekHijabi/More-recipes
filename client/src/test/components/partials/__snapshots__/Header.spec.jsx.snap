// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeHeader should render correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light"
  >
    <Link
      className="navbar-brand"
      href="/"
      replace={false}
      to="/"
    >
      <i
        className="fa fa-eercast"
      />
      moreRecipe
    </Link>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link btn"
            href="/signin"
            replace={false}
            to="/signin"
          >
            Sign in
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link btn"
            href="/signup"
            replace={false}
            to="/signup"
          >
            Sign up
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
